### PR TITLE
Fix upgrade

### PR DESCRIPTION
**Pull request by kay0u**
## Problem
- *https://github.com/YunoHost-Apps/gitlab_ynh/issues/111*

## Solution
- *Changing the way we compare versions*
- *Fix https://github.com/YunoHost-Apps/gitlab_ynh/issues/111*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Validation
---
- [ ] **Code review** : 
- [ ] **Approval (LGTM)** :  
*Code review and approval have to be from a member of @YunoHost-Apps/apps-group*
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/gitlab_ynh%20PR112%20(Kay0u)/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

File: scripts/upgrade [hunk 1]
```diff
@@ -241,16 +241,17 @@ then
...
 	while [ "$last_version" != "$current_version" ]
 	do
 
 		current_major_version=${current_version%%.*}
 
-		# If the current version is equal to the last minor version of upgrade.$current_major_version.sh, increment the major version
-		if [ "$gitlab_version" = "$current_version" ]; then
+		source ./upgrade.d/upgrade.$current_major_version.sh
+
+		# f the current version is higher than the version stored in upgrade.$current_major_version.sh, increment the major version
```
Comment (yalh76) [inline scripts/upgrade:253]: little typo # **i**f

File: scripts/upgrade [hunk 2]
```diff
@@ -241,16 +241,17 @@ then
...
+		if dpkg --compare-versions "$gitlab_version" "le" "$current_version"; then
```
Comment (yalh76) [inline scripts/upgrade:254]: Not 100% sure of using "le"... maybe better to use "gt"
I've some difficulties to decrypt what is going on...
Just remember that the result of "dpkg --compare-versions" return code is 0 if the sentence is true and the return code is not null if the sentence is false

Comment (kay0u) [reply scripts/upgrade:255]: Here `gitlab_version` is the version of the file `./upgrade.d/upgrade.$current_major_version.sh`. If this version is less than or equal to `current_version`, then this update doesn't have to be done, we can go ahead :-)

File: scripts/upgrade (hunk 1 above)
Comment (kay0u) [reply scripts/upgrade]: Thank you!

**Comment by kay0u** [comment]
Merging in 3 days!

File: scripts/upgrade (hunk 2 above)
Comment (kay0u) [reply scripts/upgrade:255]: Thank you for your review!

**Pull request merged**